descriptive-link-5effbb.md

```diff
@@ -0,0 +1,179 @@
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- [2.4.4 Link Purpose (In Context)](http://www.w3.org/TR/2008/REC-WCAG20-20081211/#navigation-mechanisms-refs)
+- [G91: Providing link text that describes the purpose of a link](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G91)
+- [G53: Identifying the purpose of a link using link text combined with the text of the enclosing sentence](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G53)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Link text describes the purpose of the link
+
+```html
+<a href="#desc">See the description of this product</a> 
+```
+
+#### Passed Example 2
+
+Accessible name describes the purpose of the link
+
+```html
+<a href="#main" aria-label="Go to the main content of the page"></a>
+```
+
+#### Passed Example 3
+
+Link text together with its context describes the purpose of the link
+
+```html
+<a href="#desc">Click</a> to see the description of this product 
+```
+
+#### Passed Example 4
+
+Both link text together with its context and accessible name describe the purpose of the link
+
+```html
+<a href="#" aria-label="Order item">Place item</a><span aria-hidden="true"> in shopping cart</span>
+```
+
+#### Passed Example 5
+
+Accessible name describes the purpose of the link
+
+```html
+<div role="link" aria-label="Skip to main content"></div>
+```
+
+#### Passed Example 6
+
+Link text describes the purpose of the link
+
+```html
+<span role="link">See description of the project</span>
+```
+
+
+### Failed
+
+#### Failed Example 1
+
+Link text does not describe the purpose of the link
+
+```html
+<a href="#desc">More</a>
+```
+
+#### Failed Example 2
+
+Accessible name does not describe the purpose of the link
+
+```html
+<a href="#main" aria-label="Go"></a>
+```
+
+#### Failed Example 3
+
+Link text does not describe the purpose of the link
+
+```html
+<div role="link">More</div>
+```
+
+#### Failed Example 4
+
+Accessible name does not describe the purpose of the link
+
+```html
+<div role="link" aria-labelledby="id1"></div>
+<div id="id1">Go</div>
+```
+
+
+### Inapplicable 
+
+#### Inapplicable Example 1
+
+`<a>` with its role changed from link to another role
+
+```html
+<a href="http://www.w3.org/WAI" role="button">Web Accessibility Initiative (WAI)</a>
+```
+
+#### Inapplicable Example 2
+
+Link that is not visible and does not have an accessible name
+
+```html
+<a href="http://www.w3.org/WAI" style="display: none;"><img src="#" /></a>
+```
+
+#### Inapplicable Example 3
+
+Link with empty accessible name
+
+```html
+<a href="http://www.w3.org/WAI"><img src="#" alt="" /></a>
+```
+
+#### Inapplicable Example 4
+
+Link with empty accessible name
+
+```html
+<a href="http://www.w3.org/WAI"><img src="#" title=""/></a>
+```
+
+#### Inapplicable Example 5
+
+Link with empty accessible name
+
+```html
+<a href="http://www.w3.org/WAI"><img src="#" aria-labelledby="id1" /></a>
+```
```

Comment:
Just a suggestion: 
Would it be worth adding a test case for an `<a>` element without a `href` attribute? Then it doesn't have the role of `link`.